docs/recipes/cloud-run.md

```diff
@@ -0,0 +1,112 @@
+---
+title: Cloud Run
+lang: en-US
+meta:
+  - name: keywords
+    content: pomerium identity-access-proxy gcp google iap serverless cloudrun 
+description: >-
+  This guide covers how to deploy Pomerium to Cloud Run and use it to protect other endpoints via Authorization Headers.
+---
+
+# Securing Cloud Run endpoints
+
+This recipe's sources can be found [on github](https://github.com/pomerium/pomerium/tree/master/examples/cloudrun)
+
+## Background
+
+Services on Cloud Run and other Google Cloud serverless products can be restricted to only permit access with a properly signed [bearer token](https://cloud.google.com/run/docs/authenticating/service-to-service).  This allows requests from other services running in GCP or elsewhere to be securely authorized despite the endpoints being public.
+
+These bearer tokens are not easily set in a browser session and must be refreshed on a regular basis, preventing them from being useful for end user authorization.  Pomerium, however, can generate compatible tokens on behalf of end users and proxy the request to these services.  
+
+## How it works
+
+- Add an IAM policy delegating `roles/run.invoker` permissions to a service account
+- Run Pomerium with access to a key for the corresponding service account
+- Publish DNS records for each protected application pointing to Pomerium
+- Configure Pomerium with appropriate policy and `enable_google_cloud_serverless_authentication`
+
+The protected application delegates trust to a GCP service account which Pomerium runs as, and Pomerium performs user based authorization on a per route basis.  This turns Pomerium into a bridge between a user-centric and service-centric authorization models.
+
+## Pre-requisites
+
+This guide assumes you have Editor access to a Google Cloud project which can be used for isolated testing, and a DNS zone which you are also able to control.  DNS does not need to be inside Google Cloud for the example to work.
+
+## Set Up
+
+To deploy Pomerium to Cloud Run securely and easily, a special image is maintained at `gcr.io/pomerium-io/pomerium-[version]-cloudrun`.  It allows sourcing configuration from GCP Secrets Manager, and sets some defaults for Cloud Run to keep configuration minimal.  We will be leveraging it in this example to store IdP credentials.
```

Comment:
Maybe link to the docker definition?

Reply:
Added a bit more context here with both links and some more detail.